src/core/manager-json.c

```diff
@@ -0,0 +1,202 @@
+/* SPDX-License-Identifier: LGPL-2.1-or-later */
+
+#include <sys/prctl.h>
+
+#include "build.h"
+#include "confidential-virt.h"
+#include "json-util.h"
+#include "manager-json.h"
+#include "manager.h"
+#include "rlimit-util.h"
+#include "syslog-util.h"
+#include "taint.h"
+#include "version.h"
+#include "virt.h"
+#include "watchdog.h"
+
+static int rlimit_build_json(sd_json_variant **ret, const char *name, void *userdata) {
+        struct rlimit *rl = userdata, buf = {};
+
+        assert(name);
+        assert(ret);
+
+        if (rl)
+                buf = *rl;
+        else {
+                const char *p;
+                int z;
+
+                /* Skip over any prefix, such as "Default" */
+                assert_se(p = strstrafter(name, "Limit"));
+
+                z = rlimit_from_string(p);
+                assert(z >= 0);
+
+                (void) getrlimit(z, &buf);
+        }
+
+        /* rlim_t might have different sizes, let's map RLIMIT_INFINITY to UINT64_MAX, so that it is the same
+         * on all archs */
+        return sd_json_buildo(ret,
+                        SD_JSON_BUILD_PAIR_UNSIGNED("soft", buf.rlim_cur),
+                        SD_JSON_BUILD_PAIR_UNSIGNED("hard", buf.rlim_max));
+}
+
+static int manager_context_build_json(sd_json_variant **ret, const char *name, void *userdata) {
+        Manager *m = ASSERT_PTR(userdata);
+        _cleanup_strv_free_ char **taints = NULL;
+
+        taints = taint_strv();
+        if (!taints)
+                return -ENOMEM;
+
+        return sd_json_buildo(ASSERT_PTR(ret),
+                        SD_JSON_BUILD_PAIR_STRING("version", GIT_VERSION),
```

Comment:
as mentioned elsewhere: i'd use upper CamelCase here, i.e. uppercase the first char, to match the D-Bus names closely.

for stuff we only create in JSON let's do lower camelCase, since that appears to be what people do there mostly.

but the stuff we already exposed is kinda all over the place, as you noticed.